### PR TITLE
Add Mixtral support - Model Provider

### PR DESCRIPTION
**Pull request by sbhavani**
# What does this PR do ?

Adds Mixtral (8x7B and 8x22B) MoE model support - Model Provider + Bridge for checkpoint conversion.

**Note**: Part 1 of 3 PRs splitting Mixtral support: (1) Model provider + bridge, (2) Pretrain recipes, (3) Finetune recipes.

# Changelog

- Add `MixtralBridge` 
- Add `MixtralModelProvider`
- Export in models `__init__.py`

## Test Suite
- Add unit tests for `MixtralBridge` (19 tests)
- Add unit tests for `MixtralModelProvider` (23 tests)
- Add functional tests for checkpoint conversion with TP/PP

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to https://github.com/NVIDIA-NeMo/Megatron-Bridge/issues/144
- Based on https://github.com/ISEEKYAN/mbridge/blob/main/mbridge/models/mixtral.py

## Release Notes

* **New Features**
  * Added support for Mixtral Mixture of Experts (MoE) models, including 8x7B and 8x22B variants, with full conversion capability to Megatron-Core format
  * Implemented comprehensive MoE configuration options with customizable expert routing, load-balancing, and parallelism support

* **Tests**
  * Added unit and functional tests validating Mixtral model conversion and configuration accuracy

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/megatron/bridge/models/mixtral/mixtral_provider.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    def __post_init__(self):
+        """Initialize computed fields after dataclass initialization."""
+        # Explicitly set kv_channels before calling parent __post_init__
+        # This is needed because dataclass field ordering can prevent parent's
+        # automatic initialization from working
+        if self.kv_channels is None:
+            self.kv_channels = self.hidden_size // self.num_attention_heads
+
+        # Call parent __post_init__ for other initialization
+        super().__post_init__()
```
Comment (ananthsub) [inline src/megatron/bridge/models/mixtral/mixtral_provider.py]: could you clarify what error you were seeing without this? what error were you seeing with the megatron core transformerconfig post init?

Comment (sbhavani) [reply src/megatron/bridge/models/mixtral/mixtral_provider.py]: I added this because I was testing direct instantiation then accessing `kv_channels`. But if we always use AutoBridge then this isn't needed.

Comment (ananthsub) [reply src/megatron/bridge/models/mixtral/mixtral_provider.py]: for the model provider, we added the `finalize()` API so that users can apply overrides to the TransformerConfig and not worry about linked initializations from `__post_init__`. 

for training, this is automatically applied. for the bridge, when using `to_megatron_model` it's also called automatically. but if using AutoBridge + to_megatron_provider, and then applying overrides, we currently need the user to call `finalize()` before initializing the model.

Comment (sbhavani) [reply src/megatron/bridge/models/mixtral/mixtral_provider.py]: I'll remove this and any test that directly accesses kv_channels

File: src/megatron/bridge/models/mixtral/mixtral_provider.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    # Embedding
+    share_embeddings_and_output_weights: bool = False
+    vocab_size: int = 32000
+
+    # MoE specific parameters
+    num_moe_experts: int = 8
+    moe_router_topk: int = 2
+    moe_ffn_hidden_size: int = 14336
+    moe_aux_loss_coeff: float = 0.01
+    moe_router_pre_softmax: bool = True
```
Comment (ananthsub) [inline src/megatron/bridge/models/mixtral/mixtral_provider.py:76]: in nemo this is False? https://github.com/NVIDIA-NeMo/NeMo/blob/5fe36df5169f96cead59a965cb1971e8f309e4a5/nemo/collections/llm/gpt/model/mixtral.py#L62

Comment (sbhavani) [reply src/megatron/bridge/models/mixtral/mixtral_provider.py:76]: maybe it's not set by default? I'd prefer to keep this True since that's what's in the HF implementation and we usually set this for other MoE models

**Review by ananthsub — DISMISSED** [review]

**Review by ananthsub — DISMISSED** [review]

File: src/megatron/bridge/models/mixtral/__init__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+
+from megatron.bridge.models.mixtral.mixtral_bridge import MixtralBridge  # noqa: F401
+from megatron.bridge.models.mixtral.mixtral_provider import MixtralModelProvider
+
+
+__all__ = [
+    "MixtralModelProvider",
+    "MixtralBridge",
+]
```
[the remark this answers is not in the document]
Comment (sbhavani) [reply src/megatron/bridge/models/mixtral/__init__.py]: Addressed in `b1785d6a`. This change is now in the PR branch.
